tsdb/head_append.go

```diff
@@ -1244,101 +1191,86 @@ func (s *memSeries) appendHistogram(t int64, h *histogram.Histogram, appendID ui
 		s.txs.add(appendID)
 	}
 
-	return true, chunkCreated
+	if nc == nil { // sample was appeneded to existing chunk or is the first sample in a new chunk
+		c.maxTime = t
+		return true, chunkCreated
+	}
+
+	if recoded { // needed to recode the chunk
+		c.maxTime = t
+		c.chunk = nc
+		return true, false // @krajorama I think technically this is lying to the caller, since we do allocate a new chunk in memory currently, but it's what the original code did
```

Comment:
I don't want to rename from created, because that would imply changing the metric as well, so I'll just explain in the docstring for the function. Updated, please take a look.